### DISCUSSION TITLE
PARCELS & Coastal NEMO applications

### DISCUSSION THREAD

**Discussion by ocgabs** [Parcels-code/Parcels#1014] [Q&A]
Is the current version of Parcels ready to work with hybrid sigma-z-partial-step levels, and enveloping bathymetry (variable number of vertical levels) like those used in coastal applications of NEMO i.e. AMM60 https://doi.org/10.1002/2017JC012960
If so, how does the type of vertical discretization is passed on to Parcels?

**Comment by reint-fischer** [comment]
Hi @ocgabs,
Yes, it is possible to use Parcels with s-grids, see also section 3.2 in the [Parcels 2.0 paper](https://gmd.copernicus.org/articles/12/3571/2019/gmd-12-3571-2019.html), although testing and examples of use cases are sparse since they are not used here in Utrecht.  If the '`depth'` dimension you pass to the fieldset is a 3D array instead of 1D, parcels creates an S-grid!

Note that there is still some uncertainty about if the vertical advection works correctly with respect to the sigma-layers and the z-layers (#759 #765 )

**Comment by ocgabs** [comment]
Thanks for your quick reply @reint-fischer,
I know the paper but it only covers S-grids, not hybrid sigma-z https://doi.org/10.5281/zenodo.3228088  (similar to HYCOM) or sigma/sigma-stretched (page 690 Guihou et al, link above) which are the more common configurations on high resolution coastal NEMO applications. The vertical coordinates are actually 4D, they evolve over time as does the sea-surface, these I have incorporated into LTRANSforNEMO (soon to be released).
Now the enveloping bathymetry (also described in the paper) is like a masking on steep bathymetry to avoid steep gradients that generate spurious currents. It generates a variable number of vertical levels, would PARCELS have issues if 'depth' is a 3D masked array? I know the answer, try and see...  
This _AdvectionRK4_3D_alternative_ is the "normal" behavior of particles one will expect in sigma coordinates, thanks for the pointer.  
Do you know if there are existing kernels for vertical diffusion?, again is not trivial in sigma coordinates, much less on hybrid ones, due to spurious accumulation of particles.

**Comment by reint-fischer** [comment]
Hi @ocgabs,
Sorry for the confusion about the grids. Time-evolving grids are definitely possible as shown in this [example](https://github.com/OceanParcels/parcels/blob/master/parcels/examples/example_dask_chunk_OCMs.py#L81-L102). 

We are not aware of anyone trying to use the enveloping bathymetry, but [partial or shaved](https://rmets.onlinelibrary.wiley.com/doi/full/10.1002/asl.187) cells work in parcels as shown [here](https://github.com/OceanParcels/parcels/blob/master/tests/test_grids.py#L557-L607). Although I only know those to be used in Z-grids, maybe you can rewrite the depth levels to include the bathymetry in the form of partial cells?

Unfortunately we also don't have any knowledge of existing kernels for vertical diffusion in sigma coordinates.
